spark/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -251,12 +252,14 @@ public String toString() {
       @Override
       public FileAppender<InternalRow> newAppender(OutputFile file, FileFormat fileFormat) {
         Schema schema = spec.schema();
+        MetricsSpec metricsSpec = MetricsSpec.fromProperties(properties);
         try {
           switch (fileFormat) {
             case PARQUET:
               return Parquet.write(file)
                   .createWriterFunc(msgType -> SparkParquetWriters.buildWriter(schema, msgType))
                   .setAll(properties)
+                  .metricsSpec(metricsSpec)
```

Comment:
I think this should be calling `forTable` instead of using `setAll`. Then `forTable` should forward properties as well as set the metrics spec.

Reply:
Does this mean that `WriterFactory` should be refactored to accept `Table` instead of individual params? This would mean that `Table` has to be serializable.

Reply:
You're right, that would require passing the table, which is probably why we don't do this. Then this is fine as you have it.

We don't want to serialize the table everywhere. That's difficult and there is no guarantee that `TableOperations` is serializable.